src/main/java/org/spongepowered/api/advancement/criteria/trigger/TriggerTypes.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.advancement.criteria.trigger;
+
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+/**
+ * An enumeration of all the available {@link TriggerType}s in minecraft.
+ */
+public final class TriggerTypes {
+
+    // SORTFIELDS:ON
+
+    public static final TriggerType ARBITRARY_PLAYER_TICK = DummyObjectProvider.createFor(TriggerType.class, "ARBITRARY_PLAYER_TICK");
+
+    public static final TriggerType BRED_ANIMALS = DummyObjectProvider.createFor(TriggerType.class, "BRED_ANIMALS");
+
+    public static final TriggerType BREWED_POTION = DummyObjectProvider.createFor(TriggerType.class, "BREWED_POTION");
+
+    public static final TriggerType CONSTRUCT_BEACON = DummyObjectProvider.createFor(TriggerType.class, "CONSTRUCT_BEACON");
+
+    public static final TriggerType CONSUME_ITEM = DummyObjectProvider.createFor(TriggerType.class, "CONSUME_ITEM");
+
+    public static final TriggerType CURSE_ZOMBIE_VILLAGER = DummyObjectProvider.createFor(TriggerType.class, "CONSUME_ITEM");
```

Comment:
"CONSUME_ITEM" again ?

Reply:
Typically, we use labels like Dummy to mean something that does absolutely nothing.  But here you're using it to do something meaningful.  Why not maker ObjectProvider generic   FooObjectProvider seems to be a factory.  See also, Design Patterns, GoF.  etc..